tests/test_views.py

```diff
@@ -974,6 +979,56 @@ def _login(self):
         """
         assert self.client.login(username=self.user.username, password="QWERTY")
 
+    def _setup_course_catalog_client(self, client_mock):
+        """
+        Sets up the Course Catalog API client
+        """
+        client = client_mock.return_value
+        client.get_course_run.return_value = {
+            'level_type': 'Type 1',
+        }
+
+    def _setup_organizations_client(self, client_mock, data=DUMMY_PLACEHOLDER, side_effect=None):
```

Comment:
@replaceafill There's only one instance where client code is passing custom values for `data` and `side_effect` to this method: https://github.com/replaceafill/edx-enterprise/blob/d7ef90e2816ec74d997f2dad09e39d62aef8ead6/tests/test_views.py#L1806. I think you should move any custom logic there, and only call this method from tests that need an identical setup for the organizations client. That will allow you to get rid of the `data` and `side_effect` parameters here:

```python
    def _setup_organizations_client(self, client_mock):
        """
        Sets up the Organizations API client
        """
        logo_mock = mock.MagicMock()
        logo_mock.url = 'logo.png'
        data = {
            'logo': logo_mock,
            'name': 'Organization',
        }
        client_mock.get_organization.return_value = data
```

Below:

```python
    import inspect

    # ...
    @ddt.data(None, ValueError)
    def test_get_course_enrollment_page_organization_errors(
            self,
            organizations_data,
            # ...
            organizations_helpers_mock,
            # ...
    ):
        # ...

        # Set up organizations API client
        if inspect.isclass(organizations_data) and issubclass(organizations_data, Exception):
            organizations_helpers_mock.get_organization.side_effect = organizations_data
        else:
            organizations_helpers_mock.get_organization.return_value = organizations_data
```
```